app/models/queue_item.rb

```diff
@@ -5,13 +5,28 @@ class QueueItem < ActiveRecord::Base
   delegate :category, to: :video
   delegate :title, to: :video, prefix: :video
 
+  validates_numericality_of :position, {only_integer: true}
+  
   def rating
-    review = Review.find_by(user_id: user.id, video_id: video.id)
     review.rating if review
   end
 
+  def rating=(new_rating)
+    if review
+      review.update_column(:rating, new_rating)
+    else
+      review = Review.new(user: user, video:video, rating: new_rating)
+      review.save(validate: false)
+    end
+  end
+
   def category_name
     category.name
   end
 
+  private
+
+  def review
+    @review ||= Review.where(user_id: user.id, video_id: video.id).first
```

Comment:
In Rails 4, `.find_by` was introduced to search for single records. Videos were recorded in Rails 3 era, where `.where(args).first` was equivalent for single record search. However `.find_by` is more efficient and idiomatic now. You should use that, and for collection searching still use `.where`.